[WIP] Cluster creation, management, and link updates for Standard and Advanced (#18327)
cockroachdb/docs: src/current/cockroachcloud/create-an-advanced-cluster.md

---
title: Create a CockroachDB Advanced Cluster
summary: Learn how to create your CockroachDB Advanced cluster.
toc: true
docs_area: deploy
---

{% include cockroachcloud/filter-tabs/create-cluster-cloud.md %}

> This page walks you through the process of creating a CockroachDB {{ site.data.products.advanced }} cluster using the [Cloud Console](httrps://cockroachlabs.cloud). To use the Cloud API instead, refer to [Create a New Cluster]({% link cockroachcloud/cloud-api.md %}#create-a-new-cluster).

Reviewer: ```suggestion
This page guides you through the process of creating a CockroachDB {{ site.data.products.advanced }} cluster using the [Cloud Console](httrps://cockroachlabs.cloud). To use the Cloud API instead, refer to [Create a New Cluster]({% link cockroachcloud/cloud-api.md %}#create-a-new-cluster).
```

    A multi-region cluster contains at minimum three regions and can survive the loss of a single region. Refer to [Planning your cluster](plan-your-cluster.html?filters=advanced) for the configuration requirements and recommendations for CockroachDB {{ site.data.products.advanced }} clusters.

1. Select the number of nodes:
    - For single-region application development and testing, you may create a 1 node cluster.
    - For single-region production deployments, we recommend a minimum of 3 nodes. The number of nodes also depends on your storage capacity and performance requirements. See [Example]({% link cockroachcloud/plan-your-cluster.md %}#example) for further guidance.
    - For multi-region deployments, we require a minimum of 3 nodes per region. For best performance and stability, you should use the same number of nodes in each region.

>     - See [Plan a CockroachDB Cloud cluster](plan-your-cluster.html) for the requirements and recommendations for CockroachDB {{ site.data.products.advanced }} cluster configuration.

Reviewer: ```suggestion
    - Refer to [Plan a CockroachDB Cloud cluster](plan-your-cluster.html) for the requirements and recommendations for CockroachDB {{ site.data.products.advanced }} cluster configuration.
```

        {% include cockroachcloud/nodes-limitation.md %}

>         Currently, you can add a maximum of 150 nodes to your cluster. For larger configurations, [contact us](https://support.cockroachlabs.com/hc/requests/new).

Reviewer: ```suggestion
        Currently, you can add a maximum of 150 nodes to your cluster. For larger configurations, [contact your Cockroach Labs account team](https://support.cockroachlabs.com/hc/requests/new).
```

Click **Next: Capacity**.

## Step 4. Enable VPC Peering (optional)

You can use [VPC peering]({% link cockroachcloud/network-authorization.md %}#vpc-peering) to connect a GCP application to a CockroachDB {{ site.data.products.cloud }} cluster deployed on GCP. A separate VPC Peering connection is required for each cluster.

> VPC peering is only available for GCP clusters. For clusters deployed on AWS, you can [set up AWS PrivateLink]({% link cockroachcloud/network-authorization.md %}#aws-privatelink) after creating your cluster. [Azure Virtual Network Peering](https://learn.microsoft.com/azure/virtual-network/virtual-network-peering-overview) is not yet supported. Refer to [CockroachDB {{ site.data.products.advanced }} on Azure]({% link cockroachcloud/cockroachdb-advanced-on-azure.md %}).

Reviewer: ```suggestion
VPC peering is available only for GCP clusters. For clusters deployed on AWS, you can [configure AWS PrivateLink]({% link cockroachcloud/network-authorization.md %}#aws-privatelink) after creating your cluster. [Azure Virtual Network Peering](https://learn.microsoft.com/azure/virtual-network/virtual-network-peering-overview) is not yet supported. Refer to [CockroachDB {{ site.data.products.advanced }} on Azure]({% link cockroachcloud/cockroachdb-advanced-on-azure.md %}).
```

You can use CockroachDB {{ site.data.products.cloud }}'s default IP range and size (`172.28.0.0/14`) as long as it doesn't overlap with the IP ranges in your network. Alternatively, you can configure the IP range:

1. In the **VPC Peering section**, select **Configure the IP range** to configure your own IP range.
    
1. Enter the IP range and size (in CIDR format) for the CockroachDB {{ site.data.products.cloud }} network based on the following considerations:
      -  As per [GCP's overlapping subnets restriction](https://cloud.google.com/vpc/docs/vpc-peering#restrictions), configure an IP range that doesn't overlap with the IP ranges in your application network.
      - The IP range and size cannot be changed after the cluster is created. Configuring a smaller IP range size may limit your ability to expand into multiple regions in the future. We recommend configuring an IP range size of `/16` or lower.

        {{site.data.alerts.callout_info}}
        Custom IP ranges are temporarily unavailable for multi-region clusters.
        {{site.data.alerts.end}}

1. Click **Next: Capacity**.

>         Once your cluster is created, see [Establish VPC Peering or AWS PrivateLink]({% link cockroachcloud/connect-to-your-cluster.md %}#establish-gcp-vpc-peering-or-aws-privatelink) to finish setting up VPC Peering for your cluster.

Reviewer: ```suggestion
        After your cluster is created, you can [establish VPC Peering or AWS PrivateLink]({% link cockroachcloud/connect-to-your-cluster.md %}#establish-gcp-vpc-peering-or-aws-privatelink).
```

> If you don't want to enable VPC Peering, leave the default selection of **Use the default IP range** as is and click **Next: Capacity**.

Reviewer: I would swap this sentence with the previous one as an "You can do this later instead" CTA. WDYT?

Author: Will update content in April

    Buffer | Additional buffer (overhead data, accounting for data growth, etc.). If you are importing an existing dataset, we recommend you provision at least 50% additional storage to account for the import functionality.
    Compression | The percentage of savings you can expect to achieve with compression. With CockroachDB's default compression algorithm, we typically see about a 40% savings on raw data size.

    For more details about disk performance on a given cloud provider, refer to: <ul><li><b>GCP</b>: <a href="https://cloud.google.com/compute/docs/disks/performance">Configure disks to meet performance requirements</a></li><li><b>AWS</b>: <a href="https://aws.amazon.com/ebs/features/#Amazon_EBS_volume_types">Amazon EBS volume types</a></li><li><b>Azure</b>: <a href="https://learn.microsoft.com/azure/virtual-machines/disks-performance">Virtual machine and disk performance</a></li></ul>

To change the hardware configuration after the cluster is created, see [Manage a CockroachDB {{ site.data.products.advanced }} Cluster]({% link cockroachcloud/cluster-management.md %}).

> See the [Example]({% link cockroachcloud/plan-your-cluster.md %}#example) for further guidance.

Reviewer: ```suggestion
Refer to [Plan your cluster]({% link cockroachcloud/plan-your-cluster.md %}#example) for examples and further guidance.
```

Click **Next: Security**.

## Step 6. Configure advanced security features

> You can add advanced security features for PCI and HIPAA [compliance]({% link cockroachcloud/compliance.md %}) at an additional cost. 

Reviewer: ```suggestion
You can enable advanced security features for PCI DSS and HIPAA [compliance]({% link cockroachcloud/compliance.md %}) at an additional cost. 
```
It must be "PCI DSS". Otherwise you could be talking about a PCI peripheral rather than Payment Card Industry.

    {{site.data.alerts.callout_danger}}
    This configuration cannot be changed after cluster creation.
    {{site.data.alerts.end}}

## Step 7. Enter billing details

1. On the **Finalize** page, verify your selections for the cloud provider, region(s), number of nodes, and the capacity.

>         Once your cluster is created, see [Establish VPC Peering or AWS PrivateLink]({% link cockroachcloud/connect-to-your-cluster.md %}#establish-gcp-vpc-peering-or-aws-privatelink) to finish setting up VPC Peering for your cluster.

Reviewer: ```suggestion
        Once your cluster is created, you can [establish VPC Peering or AWS PrivateLink]({% link cockroachcloud/connect-to-your-cluster.md %}#establish-gcp-vpc-peering-or-aws-privatelink).
```

1. Verify the hourly estimated cost for the cluster. The cost displayed does not include taxes.

    You will be billed monthly.

1. Add your preferred [payment method]({% link cockroachcloud/billing-management.md %}).
1. [If applicable]({% link cockroachcloud/frequently-asked-questions.md %}#how-do-cockroachdb-free-trials-work), the 30-day trial code is pre-applied to your cluster.

>       {{site.data.alerts.callout_info}}

Reviewer: ```suggestion

      {{site.data.alerts.callout_info}}
```

>       Make sure that you [delete your trial cluster]({% link cockroachcloud/cluster-management.md %}#delete-cluster) before the trial expires. Your credit card will be charged after the trial ends. You can check the validity of the code on the [Billing]({% link cockroachcloud/billing-management.md %}) page.

Reviewer: ```suggestion
      Remember to [delete your trial cluster]({% link cockroachcloud/cluster-management.md %}#delete-cluster) before the trial expires. Otherwise, your credit card will be charged after the trial ends. You can check the validity of the code on the [Billing]({% link cockroachcloud/billing-management.md %}) page.
```

      {{site.data.alerts.end}}

## Step 7. Name the cluster

> The cluster is automatically given a randomly-generated name. If desired, change the cluster's name. The cluster name must be 6-20 characters in length, and can include lowercase letters, numbers, and dashes (but no leading or trailing dashes). A cluster's name cannot be edited after it is created.

Reviewer: ```suggestion
The cluster is automatically given a randomly-generated name. If desired, change the cluster's name. The cluster name must be 6-20 characters in length, and can include lowercase letters, numbers, and dashes (but no leading or trailing dashes). A cluster's name cannot be edited after the cluster is created.
```

{{site.data.alerts.callout_danger}}
Testing releases, including Pre-Production Preview releases, are provided for testing and experimentation only, and are not qualified for production environments and not eligible for support or uptime SLA commitments.
{{site.data.alerts.end}}

If you install a Pre-Production Preview release, it will be upgraded to each subsequent beta or RC release automatically, before being upgraded to the GA and subsequent patch releases as they become available.

Click **Create cluster**. Your cluster will be created in approximately 20-30 minutes.

## What's next

To start using your CockroachDB {{ site.data.products.cloud }} cluster, see the following pages:

- [Connect to your cluster]({% link cockroachcloud/connect-to-your-cluster.md %})
- [Authorize users]({% link cockroachcloud/managing-access.md %})
- [Deploy a Python To-Do App with Flask, Kubernetes, and CockroachDB {{ site.data.products.cloud }}]({% link cockroachcloud/deploy-a-python-to-do-app-with-flask-kubernetes-and-cockroachcloud.md %})

If you created a multi-region cluster, it is important to carefully choose:

> - The right [survival goal](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/multiregion-survival-goals) for each database.

Reviewer: ```suggestion
- The most appropriate [survival goal](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/multiregion-survival-goals) for each database.
```

> - The right [table locality](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/table-localities) for each of your tables.

Reviewer: ```suggestion
- The most appropriate [table locality](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/table-localities) for each of your tables.
```

> Not doing so can result in unexpected latency and resiliency.  For more information, see the [Multi-Region Capabilities Overview]({% link {{ site.current_cloud_version}}/multiregion-overview.md %}).

Reviewer: ```suggestion
Otherwise, your cluster may experience unexpected latency and reduced resiliency.  For more information, refer to [Multi-Region Capabilities Overview]({% link {{ site.current_cloud_version}}/multiregion-overview.md %}).
```

{% comment %}
### [WIP] Select hardware configuration based on performance requirements

> Let's say we want to run a TPC-C workload with 500 warehouses on a CockroachDB {{ site.data.products.cloud }} cluster.

Reviewer: Ack that this is WIP, but it could use a rewrite to be somewhat less colloquial before uncommenting it.

Author: Not sure if this is still relevant
